### PR TITLE
feat(models): modify User model for out needs

### PR DESCRIPTION
**Pull request by open-dynaMIX**
This commit transforms the `User` model from the cookiecutter into the
`User` model we need (i.e. no auth, more fields).

### CODE REVIEW

File: emeis/core/models.py [hunk 1]
```diff
@@ -1,7 +1,127 @@
...
+        _("username"),
+        max_length=150,
+        unique=True,
+        help_text=_(
+            "Required. 150 characters or fewer. Letters, digits and @/./+/-/_ only."
+        ),
+        validators=[username_validator],
+        error_messages={"unique": _("A user with that username already exists.")},
+    )
+    first_name = models.CharField(_("first name"), max_length=30, blank=True)
```
Comment (winged) [inline emeis/core/models.py:51]: I wouldn't artificially limit the first name. While it's unlikely in our countries, in other regions it might be normal to have a longer first name

File: emeis/core/models.py [hunk 2]
```diff
@@ -1,7 +1,127 @@
...
+    Defined as Caluma default
+    """
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    class Meta:
+        abstract = True
+
+
+class User(UUIDModel):
```
Comment (winged) [inline emeis/core/models.py:38]: Will the API allow creating users with an already-known PK? For example if our IDP is Keycloak, will we allow the UUID to keep in sync or will we always use a separate UUID? Might be worth considering: In another project, we didn't do this and always have to think about if the "user id" is the remote or the local one". 

Doesn't help of course if the remote IDP uses anyhting other than UUID for identification. Still worth to consider though :)

**Comment by open-dynaMIX** [comment]
@winged Thanks for your feedback. This PR is not ready yet. I forgot to add `WIP`.

File: emeis/core/models.py (hunk 1 above)
Comment (open-dynaMIX) [reply emeis/core/models.py]: Totally agree. I've removed the restictions on all names (username, first and last).

File: emeis/core/models.py (hunk 2 above)
Comment (open-dynaMIX) [reply emeis/core/models.py:38]: I don't see a reason why we should not allow setting the UUID.

**Comment by open-dynaMIX** [comment]
@winged Ready for review!

File: emeis/core/models.py (hunk 2 above)
Comment (winged) [reply emeis/core/models.py:39]: It's more about me being unsure if we can do this by default via DRF. Might be worth having a test for :)

File: emeis/core/factories.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 class UserFactory(DjangoModelFactory):
     first_name = Faker("first_name")
     last_name = Faker("last_name")
-    username = Faker("uuid")
+    username = Faker("uuid4")
```
Comment (winged) [inline emeis/core/factories.py:10]: Should we really use `uuid4` as a username? I think there is a `user_name` faker that might be better suitable

File: emeis/core/tests/test_user.py [hunk 4]
```diff
@@ -1,6 +1,33 @@
 from django.urls import reverse
 from rest_framework import status
 
+from ..models import User
+
+
+def test_user_model():
+    username_raw = (
```
Comment (winged) [inline emeis/core/tests/test_user.py:8]: Tests like this might be a really good use case for `hypothesis`. We should use that way more anyhow :)

File: emeis/core/factories.py (hunk 3 above)
Comment (open-dynaMIX) [reply emeis/core/factories.py]: Done!

File: emeis/core/tests/test_user.py (hunk 4 above)
Comment (open-dynaMIX) [reply emeis/core/tests/test_user.py:8]: Nice! I'm not familiar with hypothesis yet. How would you approach this in this situation? We want to test with very specific input and expect specific output. We don't want to test `self.normalize_username()`, we just want to make sure it has been called.

**Comment by open-dynaMIX** [comment]
@winged Made some more changes. The `BaseModel` had too much copy/paste from Caluma. Please have another look.

File: emeis/core/tests/test_user.py (hunk 4 above)
Comment (winged) [reply emeis/core/tests/test_user.py:8]: You could do something like this:

```python
from hypothesis import given
from hypothesis.strategies import text

@given(username=text())
 def test_user_model(username):
    username_normal = unicodedata.normalize("NFKC", username)
    # do the thing
    assert user.username == username_normal
```

This will run your test a couple of times using various usernames (Hypothesis is really good at producing edge cases too!). You could also fill insert email, first and last name from Hypothesis as well :)

Comment (open-dynaMIX) [reply emeis/core/tests/test_user.py:8]: Nice! Will try!

Comment (open-dynaMIX) [reply emeis/core/tests/test_user.py]: @winged Awesome! Works!

Comment (winged) [reply emeis/core/tests/test_user.py]: Yeah! Very cool!
